model/src/main/kotlin/OrtResult.kt

```diff
@@ -326,9 +326,16 @@ data class OrtResult(
             packageIds.any { curation.isApplicable(it) }
         }
 
+        // TODO: Implement replacing curations for a particular provider ID, without losing the curations from the
+        //       repository configuration if applicable.
+        val providerId = "Replaced"
```

Comment:
How about changing the signature of `replacePackageCurations` to take a `PackageCurationProvider` and an `id` string instead?

Reply:
I had a long fight with this particular thing, and it was IMO by far not trivial.
The callers are CLI commands. If your  proposal was implemented this would become a functional change of the CLI.

I would like to keep the CLI changes out of scope of this PR. 
I'm already working on how to best integrate the provider IDs into CLI, but this IMO is a bigger topic and deserves it's own PR.

Fine?

Reply:
> I don't think so.

Well, that dependes on the semantic you envision for ` replacePackageCurations(provider, id = "EvaluatorCommandOption")`. Could you explain this?

Anyhow, I believe it is fine to consider this out of scope of this commit,
and I still strongly prefer to draw the line at this TODO comment and let me address it in my next PR.

Reply:
I've implemented your above proposal.